### ISSUE TITLE
Allow skipShapeProps in react/no-unused-prop-types

### ISSUE THREAD

**Issue by danielkraus** [strvcom/eslint-config-javascript#21]
Why do we have `skipShapeProps: false` for this rule?

Let's say I have:
```js
Navigation.propTypes = {
 links: PropTypes.arrayOf(
    PropTypes.shape({
      url: PropTypes.string.required,
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
 categoryTitle: PropTypes.string.isRequired,
}
```
Then I get
```
[eslint] 'links.*.url' PropType is defined but prop is never used (react/no-unused-prop-types)
```

I think we should set `skipShapeProps` to `true`. Or am I missing something? I found the similar issue that describes the problem more for example here https://github.com/yannickcr/eslint-plugin-react/issues/819.

**Comment by robertrossmann** [comment]
@dannytce indeed, why? I thought skipping any type of props validation is not a good thing but if you guys have valid reasons to turn that option off I'm okay with it.

**Comment by robertrossmann** [comment]
**ping** @dannytce.

**Comment by petrhanak** [comment]
@robertrossmann Rule `react/no-unused-prop-types` is broken. It should be set to only warning level with `skipShapeProps: true` as it is causing a lot of false positives.

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unused-prop-types.md#known-issueslimitations

Details discussed here https://github.com/yannickcr/eslint-plugin-react/issues/976

**Comment by robertrossmann** [comment]
Fixed in 7.5.0 🎉
